Rakefile

```diff
@@ -21,28 +18,9 @@ LOG = Logger.new(STDERR)
 
 desc 'Load the environment'
 task :environment do
-  environment = ENV['SINATRA_ENV'] || 'development'
-  Sinatra::Base.environment = environment
-
-  Mongoid.load!('config/mongoid.yml')
-  Mongoid.logger.level = Logger::INFO
-
-  module CommentService
-    class << self;
-      attr_accessor :config
-
-      def search_enabled?
-        self.config[:enable_search]
-      end
-    end
-  end
-
-  CommentService.config = YAML.load(application_yaml).with_indifferent_access
-
-  Elasticsearch::Model.client = Elasticsearch::Client.new(host: CommentService.config[:elasticsearch_server], log: false)
-
-  Dir[File.dirname(__FILE__) + '/lib/**/*.rb'].each { |file| require file }
-  Dir[File.dirname(__FILE__) + '/models/*.rb'].each { |file| require file }
+  # Load all of app.rb, because it is too easy to introduce bugs otherwise where Rake
+  # does not have a fix or config that is added to app.rb.
+  require File.dirname(__FILE__) + '/app.rb'
```

Comment:
@clintonb @maxrothman: Although rake may not need everything in app.rb, it needed assorted code in app.rb.  I figure that rather than creating a shared file which will be difficult to manage, and still may lead to these kinds of issues, we'll just make rake load everything.

Note that reindex must have been broken for a while now.  This didn't have to do with our changes.